src/watchmaker/workers/salt.py

```diff
@@ -631,20 +632,22 @@ def process_states(self, states, exclude):
             return
 
         cmds = []
-        states = states.lower().split(',')
         salt_cmd = self.salt_state_args
 
-        if 'highstate' in states:
+        if 'highstate' in states.lower().split(','):
             self.log.info('Applying the salt "highstate"')
-            states.remove('highstate')
+            states = ','.join(' '.join(re.sub(
+                'highstate',
+                '',
+                states,
+                flags=re.IGNORECASE).split(',')).split())
             cmds.append(salt_cmd + ['state.highstate'])
 
         if states:
             self.log.info(
                 'Applying the user-defined list of states, states=%s',
                 states
             )
-            states = ','.join(states)
```

Comment:
Adding this back depends on the approach above, but would be needed if going with my suggestion.

```suggestion
            states = ','.join(states)
```